google/cloud/internal/build_info.cc.in

```diff
@@ -29,18 +29,19 @@ std::string compiler() {
 
 // NOLINTNEXTLINE(readability-identifier-naming)
 std::string compiler_flags() {
-  static char const kCompilerFlags[] =
-      R"""(@CMAKE_CXX_FLAGS@ ${CMAKE_CXX_FLAGS_${GOOGLE_CLOUD_CPP_BUILD_TYPE_UPPER}})""";
-  return kCompilerFlags;
+  static auto const* const kCompilerFlags = new auto(
+      R"""(@CMAKE_CXX_FLAGS@ ${CMAKE_CXX_FLAGS_${GOOGLE_CLOUD_CPP_BUILD_TYPE_UPPER}})""");
+  return *kCompilerFlags;
```

Comment:
`char[]` has a trivial destructor, so this didn't really buy you anything (but it probably cost you call-time synchronization and a dynamic allocation).

That said we could make the function static a `std::string`, but whether it is worth it depends on the call pattern.